testing/runner.ts

```diff
@@ -95,74 +93,110 @@ export async function getMatchingUrls(
 
   return matchingLocalUrls.concat(matchingRemoteUrls);
 }
+
+export interface RunTestModulesOptions extends RunOptions {
+  exclude?: string[];
+}
+
 /**
- * This function runs matching test files in `root` directory.
+ * Import the specified test modules and run their tests as a suite.
+ *
+ * Test modules are specified as an array of strings and can include local files
+ * or URLs.
  *
- * File matching and excluding supports glob syntax, ie. if encountered arg is
- * a glob it will be expanded using `glob` method from `fs` module.
+ * File matching and excluding support glob syntax - arguments recognized as
+ * globs will be expanded using `glob()` from the `fs` module.
  *
- * Note that your shell may expand globs for you:
- *    $ deno -A ./runner.ts **\/*_test.ts **\/test.ts
+ * Example:
  *
- * Expanding using `fs.glob`:
- *    $ deno -A ./runner.ts \*\*\/\*_test.ts \*\*\/test.ts
+ *       runTestModules(["**\/*_test.ts", "**\/test.ts"]);
  *
- *  `**\/*_test.ts` and `**\/test.ts"` are arguments that will be parsed and
- *  expanded as: [glob("**\/*_test.ts"), glob("**\/test.ts")]
+ * Any matched directory `<dir>` will expand to:
+ *   - `<dir>/**\/*_test.ts`
+ *   - `<dir>/**\/*_test.js`
+ *   - `<dir>/**\/test.ts`
+ *   - `<dir>/**\/test.js`
+ *
+ * So the above example is captured naturally by:
+ *
+ *       runTestModules(["."]);
  */
-// TODO: change return type to `Promise<void>` once, `runTests` is updated
-// to return boolean instead of exiting
-export async function main(root: string = cwd()): Promise<void> {
+// TODO: Change return type to `Promise<void>` once, `runTests` is updated
+// to return boolean instead of exiting.
+export async function runTestModules(
+  include: string[],
+  {
+    exclude = [],
+    parallel = false,
+    exitOnFail = false,
+    only = /[^\s]/,
+    skip = /^\s*$/,
+    disableLog = false
+  }: RunTestModulesOptions = {}
+): Promise<void> {
+  // TODO: Remove the concept of a root, support parent paths.
+  const root = cwd();
+  const testModuleUrls = await getMatchingUrls(include, exclude, root);
+
+  if (testModuleUrls.length == 0) {
+    if (!disableLog) {
+      console.error("No matching test modules found.");
+    }
+    return;
+  }
+
+  if (!disableLog) {
+    console.log(`Found ${testModuleUrls.length} matching test modules.`);
+  }
+
+  for (const url of testModuleUrls) {
+    await import(url);
+  }
+
+  await runTests({
+    parallel,
```

Comment:
I think it makes sense to leave out `parallel` option for now. It doesn't work as expected know anyway. Tests are not run in parallel (by separate workers) but rather concurrently (all tests are as promises by single thread and await at the end).

Reply:
This is just for completeness and consistency with `RunOptions`, the default value given to it at the top is the same as for `runTests()`. I'm just propagating it here since we're extending that.